retrieval/discovery/kubernetes/endpoints.go

```diff
@@ -89,6 +89,11 @@ func (e *Endpoints) Run(ctx context.Context, ch chan<- []*config.TargetGroup) {
 			send(e.buildEndpoints(o.(*apiv1.Endpoints)))
 		},
 		DeleteFunc: func(o interface{}) {
+			s, ok := o.(*cache.DeletedFinalStateUnknown)
+			if ok {
+				o = s.Obj
+			}
+
 			send(&config.TargetGroup{Source: endpointsSource(o.(*apiv1.Endpoints).ObjectMeta)})
```

Comment:
Yea, I think we should stick with proper validation everywhere to avoid future panics. The underlying code is not easy enough to reason about to rely on contract here as the issue has shown.

https://github.com/kubernetes/kubernetes/pull/34694/files